gradle/build-complete.gradle

```diff
@@ -0,0 +1,26 @@
+String buildNumber = System.getenv('BUILD_NUMBER')
+
+if (buildNumber) {
+    String uploadFile = "build/${buildNumber}.tar.bz2"
+    project.gradle.buildFinished { result ->
+        println "build complete, generating: $uploadFile"
+        project.sync {
+            from fileTree(projectDir)
+            include "**/*.hprof"
+            include "**/build/testclusters/**"
+            exclude "**/build/testclusters/**/data"
+            exclude "**/build/testclusters/**/distro"
+            exclude "**/build/testclusters/**/repo"
+            exclude "**/build/testclusters/**/extract"
+            exclude "build/upload-files"
+            into "build/upload-files"
+            includeEmptyDirs = false
+        }
+        if (project.file(uploadFile)) {
+            project.delete(uploadFile)
+        }
+        ant.tar(destfile: uploadFile, compression: "bzip2", longfile: "gnu") {
+            fileset(dir: "build/upload-files")
```

Comment:
Couldn't we just configure the correct filesets here instead of having the intermediary copy step? If you don't want to bother with the nasty Ant fileset stuff you can just construct a `FileTree` and add it to the Ant task via [`addToAntBuilder()`](https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileCollection.html#addToAntBuilder-java.lang.Object-java.lang.String-),

Reply:
I tried :
```
ant.tar(destfile: uploadFile, compression: "bzip2", longfile: "gnu") { builder ->
            fileTree(projectDir)
                    .include("**/*.hprof")
                    .include("**/build/testclusters/**")
                    .exclude("**/build/testclusters/**/data")
                    .exclude("**/build/testclusters/**/distro")
                    .exclude("**/build/testclusters/**/repo")
                    .exclude("**/build/testclusters/**/extract")
                    .exclude("build/upload-files")
                    .addToAntBuilder(ant, "fileset", org.gradle.api.file.FileCollection.AntType.FileSet)
        }
```
But it takes way longer, I gave up on it after 1 minute. It generated an archive 334 bytes in size during this time ...

Reply:
```
 ant.tar(destfile: uploadFile, compression: "bzip2", longfile: "gnu") { builder ->
            fileTree(projectDir)
                    .include("**/*.hprof")
                    .include("**/build/testclusters/**")
                    .exclude("**/build/testclusters/**/data")
                    .exclude("**/build/testclusters/**/distro")
                    .exclude("**/build/testclusters/**/repo")
                    .exclude("**/build/testclusters/**/extract")
                    .exclude("build/upload-files")
                    .addToAntBuilder(ant, "resources")
        }
```
Doesn't generate an archive

Reply:
Looks like the Ant tar task isn't a "true" fileset so it doesn't work the way we'd expect. I guess doing the intermediary copy is the best option here.